notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py

```diff
@@ -0,0 +1,688 @@
+        # move 0's to None to avoid divide by 0
+        lambda x: x
+        if x > 0
+        else None
+    )
+)
+
+
+fig_maturity_loc_and_age = px.scatter(
+    df_projects.sort_values(by="category"),
+    hover_name="Project Name",
+    x="Duration Created to Now in Years",
+    y="total lines of GitHub detected code (Log Scale)",
+    width=1200,
+    height=500,
+    color="category",
+    color_discrete_sequence=category_color_sequence,
+)
+
+
+# customize the chart layout
+fig_maturity_loc_and_age.update_layout(
+    title=f"Project Lines of Code and Age in Years (click categories for further focus)",
+    yaxis_title="Total Lines of Code (Log Scale)",
+    xaxis_title="Project Age (years)",
+)
+
+fig_collection.append(
+    {
+        "plot": fig_maturity_loc_and_age,
+        "description": """
+        This plot shows the lines of code (LoC) and age in years to help indicate relative project maturity.
+        Lines of code in this case are the sum total of GitHub detected programming language dialects.
+        """,
+        "findings": """
+        We can observe that many tools within the Cytomining Ecosystem landscape reside within a certain range
+        of total LoC and that projects of greater age do not necessarily have more code.
+        Tools of <code>loi-focus</code> are nearby the LoC of other tools with greater age.
+        """,
+        "section": "Maturity",
+    }
+)
+
+fig_maturity_loc_and_age.show()
+
+# +
+# scatter plot for maturity based on project
+
+# add log of github stars to help visualize
+"""df_projects["GitHub Open Issues (Log Scale)"] = np.log(
+    df_projects["GitHub Open Issues"].apply(
+        # move 0's to None to avoid divide by 0
+        lambda x: x
+        if x > 0
+        else None
+    )
+)
+"""
+
+fig_maturity_latest_commit = px.scatter(
+    df_projects.sort_values(by="category"),
+    hover_name="Project Name",
+    x="Date Most Recent Commit",
+    y="GitHub Stars (Log Scale)",
+    width=1200,
+    height=500,
+    color="category",
+    color_discrete_sequence=category_color_sequence,
+)
+
+
+# customize the chart layout
+fig_maturity_latest_commit.update_layout(
+    title=f"Project Stars and Most Recent Commit Datetime (click categories for further focus)",
+)
+
+fig_collection.append(
+    {
+        "plot": fig_maturity_latest_commit,
+        "description": """
+        This plot demonstrates project GitHub stars and most recent commit datetime as a way to observe
+        potential decay for projects which may have been popular at one point but are no longer maintained.
+        """,
+        "findings": """
+        We find that relevant open source, microscopy analysis tools, and tools of <code>loi-focus</code> are
+        all recently updated. Adjacent tools demonstrate historical usage but are somewhat less frequently maintained.
+        Many tools from GitHub queries show a lack of updates for greater than 2 years.
```

Comment:
```suggestion
        Many tools from GitHub queries show a lack of updates for greater than 2 years.
        This indicates that many GitHub projects that may re-invent Cytomining functionality are likely abandoned after the authors perform bespoke analyses.
```

True? Or too much of a stretch?

Reply:
Thanks - I think this is mostly true, but I softened the suggestions here as hypotheticals (I don't know if we have enough evidence yet solely from GitHub metrics to state as a fact).

Adjusted:
```text
Many tools from GitHub queries show a lack of updates for greater than 2 years.
We hypothesize that many GitHub projects re-invent functionality related to the Cytomining Ecosystem.
These projects are sometimes abandoned after authors perform bespoke or one-time analyses.
```

Food for thought: I think in analyzing the readme's of these projects we could ascertain the "bespokeness" of individual projects (i.e. text like "this repo relates to <single dataset> and <single methods> and <single publication>"). I imagine but don't know that we could end up with NLP-defined classes, and that one of those classes would likely be "one-time research projects" (or something of that nature).